### ISSUE TITLE
[Feature Request] Better usability:

### ISSUE THREAD

**Issue by scientes** [provinzio/CoinTaxman#24]
To make this programm accessible to a wider audience it should be considered making it usable for people who do not know how to use python/ a terminal. Maybe a simple gui/ local webpage or something like that.  And or even compilation into an executable for windows.

Note that this is not a that important feature Request, but more an idea for people who want to contribute.

**Comment by wearymanateevedaknotstabooresisting** [comment]
I created myself a web-based GUI for another tax calculation algorithm below, before I joined this project development.

So there are different options, while we should aim to support as much operating systems as possible (Linux, MacOS, Windows). In my opinion, the easiest way to achieve this is a web-based GUI. Django and its templates to trigger the basic features of CoinTaxman could be a good way to start. I am open for different suggestions!

**Comment by provinzio** [comment]
In my professional environment, I mainly work with "normal" GUIs (mostly `PySide2`, python port of Qt). This makes me a little biased towards this kind of GUIs. Using Qt might be a slight overkill. Therefore I would suggest something like `PySimpleGUI` in combination with `PyInstaller` ([might be unsupported in the future](https://github.com/pyinstaller/pyinstaller/issues/4404), we could checkout `cx_Freeze` if that worries us).

But to be honest, I don't mind providing CoinTaxman with a web gui if we you think that's smarter. (We might be able to refactor a lot of code from your project @wearymanateevedaknotstabooresisting.)

**Comment by wearymanateevedaknotstabooresisting** [comment]
Honestly, we don't do it for us, the tech savvy people who can start the program from command line. If we want to support the average user, and I think this may be a wise idea, we should aim for anything that works on as much machines with lowest effort as possible. So, if a standalone GUI is easy to install and works, that's fine - also the web GUI is fine if the web server opens with a click, I don't know if this is actually harder to ship, though.

In summary, the first valid concept which fulfills the easy-to-install, easy-to-use, easy-to-develop should be the go-to solution.

**Comment by scientes** [comment]
Web gui would be nice because that would be increadibly easy to make interactive plots using stuff like bokeh. And making a Web App from that using electron should be possible if needed.

**Comment by wearymanateevedaknotstabooresisting** [comment]
I actually tried Django with Electron and it was PITA. Can't recommend those two together. However, I know that Electron is a great way to publish nice cross-OS GUIs through easy web development.

**Comment by eric15342335** [comment]
> To make this programm accessible to a wider audience it should be considered making it usable for people who do not know how to use python/ a terminal. Maybe a simple gui/ local webpage or something like that

Idk but maybe `autopytoexe` package does that.
![resized-autopytoexe](https://user-images.githubusercontent.com/70310617/110487791-d38c8580-80e5-11eb-9c69-e11e32b64098.png)

**Comment by shredEngineer** [comment]
Having a GUI is a great idea. I find PyQt5 quite easy to use, mature, and it works without problems in Linux and Windows (& Mac I guess). Here's a very simple example (< 100 LOC) I recently created for a friend, so you can see what it looks like:

https://gist.github.com/shredEngineer/693f6f992626fc0706f19344e257c5fb

I already created a large project using PyQt5 and arranged all widgets by hand, but theres also QtDesigner to make things really simple.

FastAPI is nice and all that, but I think a plain and simple GUI would be best. Don't know about the other suggestions.

**Comment by scientes** [comment]
why not both^^ (semi serious)

it could maybe be switched on and off via a config file or be different projects wich just use Cointaxman as a module which they import. The only necessity in this project is that we restructure it so that: it is possible to use this as a module and not just as a script. I myself would just like to chug this thing onto a webserver so that i can use this from multiple Computers. 

Bit offtopic but could be helpful for that (maybe worth a new issue):
-  also it would be nice if we switch to something like sqlalchemy or [encode/orm](https://github.com/encode/orm) (sqlalchemy but asynchronus), just to open possibilites (i'd rather not do parrallelization on sqlite, possible via WAL but not the nicest thing on the earth)
- A single database would be nice not one per exchange, could be handled via prefixes (would be needed for sqlalchemy)
- and things like evaluate_taxation()  could be made into an endpoint so that i could call it for every coin individually (this would be one of the CPU bottlenecks if we are dealing with a large amount of transactions so it would be nice to be able to parallelize on a per coin basis) i mean i could just call __evaluate_taxation() but that would be not the cleanest thing in my mind

**Comment by jhoogstraat** [comment]
Maybe Flutter can provide a reasonable web and desktop app (or even mobile for that matter). It could call python via a ffi.
